### DISCUSSION TITLE
Mismatched types not throwing error?

### DISCUSSION THREAD

**Discussion by winterqt** [diwic/dbus-rs#404] [Q&A] [answered]
In the following snippet:
```rust
use dbus::blocking::Connection;
use std::time::Duration;

fn main() {
    let conn = Connection::new_system().unwrap();

    let proxy = conn.with_proxy(
        "org.freedesktop.systemd1",
        "/org/freedesktop/systemd1",
        Duration::from_secs(5),
    );

    let (res,): (Vec<(u32,)>,) = proxy
        .method_call("org.freedesktop.systemd1.Manager", "ListUnits", ())
        .unwrap();

    dbg!(res);
}
```

where the struct type is actually `ssssssouso`, I'd expect an error (since a `String` is trying to be parsed into a `u32`). Instead, `res` is equal to `[]`.

Why is this?

**Comment by winterqt** [comment, marked as answer]
Figured it out: this seems to be a limitation `Vec`'s implementation of `Get::get`:
```rust
impl<'a, T: Arg + Get<'a>> Get<'a> for Vec<T> {
    fn get(i: &mut Iter<'a>) -> Option<Self> {
        <Array<T, Iter<'a>>>::get(i).map(|a| a.collect())
    }
}
```

As far as I can tell, as long as the return type is an array, this will return `Some`, leading to no error being thrown by `Iter::read`.

**Comment by diwic** [comment]
There seems to be a todo item for exactly this:

    impl<'a, T: Arg + Get<'a>> Get<'a> for Array<'a, T, Iter<'a>> {
        fn get(i: &mut Iter<'a>) -> Option<Array<'a, T, Iter<'a>>> {
            i.recurse(Self::ARG_TYPE).map(|si| Array(si, PhantomData))
            // TODO: Verify full element signature?
        }
    }

PRs welcome. :-)
